File: create_post.rs
use std::fs::File;
use std::io;
use std::io::Write;
use std::path::Path;
use std::process::Command;

fn main() {
    println!("Title of the publication:");

    let mut title = String::new();
    io::stdin()
        .read_line(&mut title)
        .expect("Failed to read line.");
    title = title.trim().to_string();
    let fname = title.to_lowercase().replace(" ", "-") + ".md";

    let path = Path::new("content/archive").join(fname);
    let display = path.display();

    let mut file = match File::create(&path) {
        Err(why) => panic!("couldn't create {}: {}", display, why),
        Ok(file) => file,
    };

    let output = format!(
        "+++\ntitle = \"{}\"\ndate = {}\n+++\n\n\n",
        title, "2023-01-01"
    );

    match file.write_all(output.as_bytes()) {
        Err(why) => panic!("couldn't write to {}: {}", display, why),
        Ok(_) => println!("successfully wrote to {}", display),
    }
    Command::new("/usr/bin/zsh")
        .arg("-c")
        .arg(format!("vim \"+call cursor(6,0)\" {}", display))
        .spawn()
        .expect("vi command failed")
        .wait()
        .expect("editor failed");
}

File: create_post.cc
#include <ctime>
#include <fstream>
#include <iostream>
#include <algorithm>



int main() {
    std::cout << "Title of the publication:\n";
    std::string title, fname, date;
    std::getline(std::cin, title);
    fname = title;
    std::transform(fname.begin(), fname.end(), fname.begin(),
            [](unsigned char c){ return std::tolower(c); });
    for (auto &t: fname) if (t == ' ') t = '-';
    fname += ".md";

    std::string path = "content/archive/" + fname;
    std::ofstream file(path);

    time_t now = time(0);
    struct tm  tstruct;
    char buf[80];
    tstruct = *localtime(&now);
    strftime(buf, sizeof(buf), "%Y-%m-%d", &tstruct);

    file << "+++\ntitle = \"" << title << "\"\ndate = " << buf << "\n+++\n\n\n";
    file.close();
    std::string call = "/usr/bin/vim \"+call cursor(6,0)\" " + path;
    system(call.c_str());
    std::cout << "File created at: " << path << std::endl;
}
